tatanka/tatanka.go

```diff
@@ -681,3 +721,25 @@ func (t *Tatanka) clientDisconnected(peerID tanka.PeerID) {
 		tt.Send(note)
 	}
 }
+
+// broadcastRates sends market rates to all fiat rate subscribers every
+// t.fiatRateBroadcastInterval. This method blocks and must be called from a
+// goroutine.
+func (t *Tatanka) broadcastRates() {
+	broadcastChan := t.fiatRateOracle.BroadcastChan()
+	for {
+		select {
+		case <-t.ctx.Done():
+			return
+		case rates := <-broadcastChan:
+			t.clientMtx.RLock()
+			defer t.clientMtx.RUnlock()
```

Comment:
You can't use the defer here because it does not happen until the function returns. However I don't think you need to hold a lock at all as `batchSend` does.

Reply:
Gotcha!